core/src/test/java/org/sonarsource/sonarlint/core/container/connected/update/ModuleHierarchyDownloaderTest.java

```diff
@@ -69,9 +72,59 @@ public void test() {
   }
 
   @Test
+  public void testNoPagination() throws IOException {
+    TreeWsResponse.Builder responseBuilder = TreeWsResponse.newBuilder()
+      .setPaging(Paging.newBuilder().setTotal(500));
+    for (int i = 0; i < 500; i++) {
```

Comment:
Here I want to test that no extra/useless pagination is made when result count is exactly the page number. It would be unnoticed at runtime since doing a WS call with a page index too big simply return an empty result list.